### ISSUE TITLE
Pi Zero 2 W, HyperPixel2.1 and the Rainbow

### ISSUE THREAD

**Issue by gregamillion** [pimoroni/hyperpixel2r#10]
I'm trying to set up the Pi Zero 2 W with the HyperPixel 2.1 Round.

I'm sure there's nothing wrong with the display but I can't get it operating. I'm hoping someone can point me in the right direction.

I've got the Pi Zero 2 W running on an external display (with the HyperPixel connected as well), so I know the RPi is functional. I'm using the power supply and sd card from a PiShop.us starter kit, so I'm confident they are each a proper fit.

I've tried both the latest Raspberry Pi OS (2023-05-03) and the Raspberry Pi OS (Legacy) Debian Buster (2023-05-03) and, in both cases, updated the system following the instructions on the PiShop site:

git clone https://github.com/pimoroni/hyperpixel2r
cd hyperpixel2r
sudo ./install.sh)

I've booted the system in both cases with and without an external HDMI display attached. In both cases when the external display was disconnected, I was able to SSH into the Raspberry Pi, interact with it and shut it down or reboot it remotely.

I have the pins seated snugly into the HyperPixel display and everything is fit comfortably with the Short Pi Standoffs.

On my latest attempt, I started from scratch with Buster again. The newly minted Buster card booted fine, the external HDMI display fired up, the HyperPixel display stayed black, and I ran the installation process as described on the site:
git clone https://github.com/pimoroni/hyperpixel2r
cd hyperpixel2r
sudo ./install.sh

When I restart, I get the rainbow screen on the external HDMI display and it stays that way. The HyperPixel display stays dark. I can still SSH into the Raspberry Pi and access raspi-config, but the external HDMI is stuck on the rainbow screen. When I reboot via SSH, I get the same result.

Any suggestions would be much appreciated. Thanks!

**Comment by gregamillion** [comment]
!UPDATE! - After stepping through every manner of config to no avail, I took my coach's advice and removed and reinstalled the display. The only other way I could see to seat the display was to invert the pins, so I did. They now stick up proud of the PiZero2W significantly, but when I connected power to the device, the backlight turned on for the first time! I'm also able to control the backlight by SSH, but I'm still not getting the GUI either in Bullseye or Buster, both with GL turned off in the config options. 

So I think this confirms that the display hardware is fine, but I'm still looking a black (if illluminated) screen. 

Thoughts?

**Comment by gregamillion** [comment]
!UPDATE! - I connected the HyperPixel2.1 to another RPi I had lying around (a RPi3, I think) and, after waiting about 5 minutes for everything to boot up, the HyperPixel2.1 came to life and displayed the GUI. So it works.
And the Pi02w is functioning, because I can hook it up to an external display and do everything I want via the GUI. 

So... can anyone who is combining these two bits of hardware tell me the version of Raspian they're using and any modifications (aside from loading the hyperpixel2 code found in this repository) they had to make to get this to work? I don't care about using the touchscreen capabilities if that makes the scenario easier.

It seems VERY CLOSE to working, but I'm missing something vital somewhere. Thanks!

**Comment by kirscfil** [comment]
@gregamillion Hey I have also struggled with this - how I got the display to work on Zero 2 W Buster is I reinstalled the library with the display attached and then rebooted. Reinstallation without the display did not result in success. Hope this helps.

Also worth noting - even with this I did not succeed in changing the orientation:

```
$ hyperpixel2r-rotate left
This rotate utility only works with the Raspberry Pi OS desktop version or X-based alternatives.

Rotation requires hardware support on the Pi 4 or Pi 400
Ensure "dtoverlay=vc4-fkms-v3d" is enabled in /boot/config.txt
```

**Comment by gregamillion** [comment]
Thanks! 
At my wits end, I ordered another Pi Zero 2 W, but this time with soldered pins. I began to suspect that the hardware connection was faulty on my current combination of Pi & HyperPixel.
I'll try your suggestion on both the new and old Pis today!
Thanks again!

**Comment by gregamillion** [comment]
@kirscfil ! Thank you! Something between the new Pi and your instructions set things right. I've got a functional HyperPixel2r on the Pi Zero 2 W with Buster (2023-05-03 release) and the update from this repository. With no other adjustments, the touch screen is operational, which I did not expect. Thanks again for your help!

(Edit for historical completeness) I didn't try @kirscfil 's solution on the first Pi Zero 2 W as I'm working on a deadline and am afraid to lose any progress.  I still suspect the hardware connection was at fault because the new Pi, with soldered pins, caused the backlight on the hyperpixel2r display to light as soon as the power was introduced. With the other one the backlight only came on sometimes and, even then, flickered occasionally.
